src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,165 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
```

Comment:
The `Node` class should be an inner class within `MyLinkedList` to encapsulate it properly. This will also allow you to make the fields private, improving encapsulation.